pr_agent/git_providers/bitbucket_provider.py

```diff
@@ -5,16 +5,25 @@
 
 import requests
 from atlassian.bitbucket import Cloud
+from starlette_context import context
 
 from ..config_loader import get_settings
 from .git_provider import FilePatchInfo, GitProvider
 
 
 class BitbucketProvider(GitProvider):
-    def __init__(self, pr_url: Optional[str] = None, incremental: Optional[bool] = False):
+    def __init__(
+        self, pr_url: Optional[str] = None, incremental: Optional[bool] = False
+    ):
         s = requests.Session()
-        s.headers['Authorization'] = f'Bearer {get_settings().get("BITBUCKET.BEARER_TOKEN", None)}'
-        s.headers['Content-Type'] = 'application/json'
+        try:
+            bearer = context.get("bitbucket_bearer_token", None)
+            s.headers["Authorization"] = f"Bearer {bearer}"
+        except Exception:
+            s.headers[
+                "Authorization"
+            ] = f'Bearer {get_settings().get("BITBUCKET.BEARER_TOKEN", None)}'
+        s.headers["Content-Type"] = "application/json"
         self.headers = s.headers
```

Comment:
**Suggestion:** Handle exceptions when getting the bearer token in the BitbucketProvider constructor.
```suggestion
        try:
            bearer = context.get("bitbucket_bearer_token", None)
            s.headers["Authorization"] = f"Bearer {bearer}"
        except Exception as e:
            logging.error(f"Failed to get bearer token from context: {e}")
            try:
                s.headers[
                    "Authorization"
                ] = f'Bearer {get_settings().get("BITBUCKET.BEARER_TOKEN", None)}'
            except Exception as e:
                logging.error(f"Failed to get bearer token from settings: {e}")
                raise e
        s.headers["Content-Type"] = "application/json"
```